src/utils/keys.ts

```diff
@@ -132,9 +133,15 @@ export abstract class Keys {
     }
   }
 
-  public static filter<T extends {}>(data: T[], matchingPatterns: string | string[]): Partial<T>[];
-  public static filter<T extends {}>(data: T, matchingPatterns: string | string[]): Partial<T>;
-  public static filter<T extends ({} | {}[])>(data: T|T[], matchingPatterns: string | string[]): Partial<T> | Partial<T>[] {
+  public static filter<T extends {}>(data: T, matchingPatterns: TWildCard | [TWildCard], allowMutateData?: boolean): T;
+  public static filter<T extends {}>(data: T[], matchingPatterns: TWildCard | [TWildCard], allowMutateData?: boolean): T[];
+  public static filter<T extends {}>(data: T[], matchingPatterns: string | string[], allowMutateData?: boolean): Partial<T>[];
+  public static filter<T extends {}>(data: T, matchingPatterns: string | string[], allowMutateData?: boolean): Partial<T>;
+  public static filter<T extends ({} | {}[])>(data: T|T[], matchingPatterns: string | string[], allowMutateData= false): Partial<T> | Partial<T>[] {
+    if (matchingPatterns === WILD_CARD || (Array.isArray(matchingPatterns) && matchingPatterns.length === 1 && matchingPatterns[0] === WILD_CARD)) {
```

Comment:
How about we extract this into a function like `isWildCardOnly()`? Thinking it would improve readability.

Reply:
Will do 👍🏻